hack/patches/unset_status_manifestval.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/vendor/github.com/manifestival/manifestival/manifestival.go b/vendor/github.com/manifestival/manifestival/manifestival.go
+index b8d07b8bc..892433fb6 100644
+--- a/vendor/github.com/manifestival/manifestival/manifestival.go
++++ b/vendor/github.com/manifestival/manifestival/manifestival.go
+@@ -133,7 +133,7 @@ func (m Manifest) apply(spec *unstructured.Unstructured, opts ...ApplyOption) er
+ 		annotate(current, v1.LastAppliedConfigAnnotation, lastApplied(current))
+ 		return m.Client.Create(current, opts...)
+ 	} else {
+-		diff, err := patch.New(current, spec)
++		diff, err := patch.New(unsetStatus(current), unsetStatus(spec))
+ 		if err != nil {
+ 			return err
+ 		}
+@@ -156,6 +156,14 @@ func (m Manifest) apply(spec *unstructured.Unstructured, opts ...ApplyOption) er
+ 	}
+ }
+ 
++func unsetStatus(spec *unstructured.Unstructured) *unstructured.Unstructured {
++	spec = spec.DeepCopy()
++	if _, ok := spec.Object["status"]; ok {
```

Comment:
Probably it will work for all types (due to different schemas) but can we make it specific to stateful sets for now?

Reply:
> I guess status should never be stored in manifest's config.

it's not coming from the manifest config, it's coming from the fact that the status struct is a concrete non-pointer type and any field in the status that is optional but it's not a pointer will be considered in the diff, so it's generally applicable to any type